packages/@aws-cdk/aws-codebuild/test/integ.aws-deep-learning-container-build-image.expected.json

```diff
@@ -0,0 +1,279 @@
+              }
+            },
+            {
+              "Action": [
+                "ecr:BatchCheckLayerAvailability",
+                "ecr:GetDownloadUrlForLayer",
+                "ecr:BatchGetImage"
+              ],
+              "Effect": "Allow",
+              "Resource": {
+                "Fn::Join": [
+                  "",
+                  [
+                    "arn:",
+                    {
+                      "Ref": "AWS::Partition"
+                    },
+                    ":ecr:",
+                    {
+                      "Ref": "AWS::Region"
+                    },
+                    ":",
+                    {
+                      "Fn::FindInMap": [
+                        "AwsDeepLearningContainersRepositoriesAccounts",
+                        {
+                          "Ref": "AWS::Region"
+                        },
+                        "account"
+                      ]
+                    },
+                    ":repository/mxnet-training"
+                  ]
+                ]
+              }
+            },
+            {
+              "Action": "ecr:GetAuthorizationToken",
+              "Effect": "Allow",
+              "Resource": "*"
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "PolicyName": "ProjectRoleDefaultPolicy7F29461B",
+        "Roles": [
+          {
+            "Ref": "ProjectRole4CCB274E"
+          }
+        ]
+      }
+    },
+    "ProjectC78D97AD": {
+      "Type": "AWS::CodeBuild::Project",
+      "Properties": {
+        "Artifacts": {
+          "Type": "NO_ARTIFACTS"
+        },
+        "Environment": {
+          "ComputeType": "BUILD_GENERAL1_LARGE",
+          "Image": {
+            "Fn::Join": [
+              "",
+              [
+                {
+                  "Fn::FindInMap": [
+                    "AwsDeepLearningContainersRepositoriesAccounts",
+                    {
+                      "Ref": "AWS::Region"
+                    },
+                    "account"
+                  ]
+                },
+                ".dkr.ecr.",
+                {
+                  "Ref": "AWS::Region"
+                },
+                ".",
+                {
+                  "Ref": "AWS::URLSuffix"
+                },
+                "/mxnet-training:1.4.1-gpu-py36-cu100-ubuntu16.04"
+              ]
+            ]
+          },
+          "ImagePullCredentialsType": "SERVICE_ROLE",
+          "PrivilegedMode": false,
+          "Type": "LINUX_GPU_CONTAINER"
+        },
+        "ServiceRole": {
+          "Fn::GetAtt": [
+            "ProjectRole4CCB274E",
+            "Arn"
+          ]
+        },
+        "Source": {
+          "BuildSpec": "{\n  \"version\": \"0.2\",\n  \"phases\": {\n    \"build\": {\n      \"commands\": [\n        \"ls\"\n      ]\n    }\n  }\n}",
+          "Type": "NO_SOURCE"
+        }
+      }
+    }
+  },
+  "Mappings": {
+    "AwsDeepLearningContainersRepositoriesAccounts": {
```

Comment:
I think the better structure for this mapping is better to look something more like this:

```
Mappings: {
  RegionInfo: {
    us-east-1: {
       dlcRepositoryAccount: 763104351884,
       elbAccount: 12345768012,
       cloudFrontHostedZoneId: AIJ35819Z
    }
  },
}
```

etc. I'm not looking forward to having a different mapping for every key, especially when mappings are already 2-level by default. 

Means that there should be some RegionInfo helper that progressively adds facts to a global map if a new fact is required in an agnostic region. I will be frank that I'm not 100% sure where that helper should live. Does `@aws-cdk/core` already depend on `region-info`? If so, that might be a good place to put it.

Reply:
I'm actually not sure about this. It would mean that everyone who uses just Deep Learning Containers will also get all information about ELB accounts, CloudFront hosted zone IDs, and a lot of different stuff they probably have zero interest in. This will:

a) make their template a lot bigger, and
b) as they upgrade their version of the CDK, every change to ELB accounts, CloudFront hosted zone IDs, etc. will result in a needless change to their CodeBuild template.

Reply:
Obviously I did not mean "just stuff all facts into a single static map that gets included into every template". Only the facts needed by the template should be in the map. To quote myself:

> Means that there should be some RegionInfo helper that **progressively adds facts** to a global map **if a new fact is required** in an agnostic region. 

On second thought, nevermind I guess. We will kick the can down the road and address this when we have 5 mappings.